rcl/src/rcl/lexer_lookahead.c

```diff
@@ -222,6 +222,11 @@ rcl_lexer_lookahead2_expect(
     return ret;
   }
   if (type != lexeme) {
+    if (RCL_LEXEME_NONE == lexeme || RCL_LEXEME_EOF == lexeme) {
```

Comment:
@sloretz I added this when I noticed that parsing argv[0] was always giving:

```
[DEBUG] [rcl]: arg 0 (/home/dhood/ros2_ws/install_isolated/demo_nodes_cpp/lib/demo_nodes_cpp/talker) error 'Expected lexeme type 19, got 1 at index 77, at /home/dhood/ros2_ws/src/ros2/rcl/rcl/src/rcl/lexer_lookahead.c:227'
```

That was the EOF case; perhaps the NONE case actually shouldn't be treated the same?

Reply:
ah, that's the difference, thanks. I tried to point it out in docs https://github.com/ros2/rcl/pull/248/files#diff-b5f802bb85e2a3c95df3c6bbfd75b826R33, but I admit to not being aware of the different ways in which they're used; if the distinction is not so simple, we can remove that comment